src/storage/memory_store.ts

```diff
@@ -0,0 +1,697 @@
+   */
+  async getToken(tokenUid: string): Promise<(ITokenData & Partial<ITokenMetadata>) | null> {
+    const tokenConfig = this.tokens.get(tokenUid);
+    if (tokenConfig === undefined) {
+      return null;
+    }
+    const DEFAULT_TOKEN_META: ITokenMetadata = {
+      numTransactions: 0,
+      balance: {
+        tokens: { unlocked: 0, locked: 0 },
+        authorities: {
+          mint: { unlocked: 0, locked: 0 },
+          melt: { unlocked: 0, locked: 0 },
+        },
+      }
+    };
+    const tokenMeta = this.tokensMetadata.get(tokenUid);
+
+    return { ...tokenConfig, ...DEFAULT_TOKEN_META, ...tokenMeta };
+  }
+
+  /**
+   * Fetch the token metadata from the storage.
+   *
+   * @param {string} tokenUid The token id to fetch metadata.
+   * @returns {Promise<ITokenMetadata | null>} The token metadata if present
+   */
+  async getTokenMeta(tokenUid: string): Promise<ITokenMetadata | null> {
+    const tokenMeta = this.tokensMetadata.get(tokenUid);
+    if (tokenMeta === undefined) {
+      return null;
+    }
+    return tokenMeta;
+  }
+
+  /**
+   * Save a token on storage
+   * @param {ITokenData} tokenConfig Token config
+   * @param {ITokenMetadata|undefined} [meta] The token metadata
+   * @async
+   * @returns {Promise<void>}
+   */
+  async saveToken(tokenConfig: ITokenData, meta?: ITokenMetadata | undefined): Promise<void> {
+    if (this.tokens.has(tokenConfig.uid)) {
+      throw new Error('Already have this token');
+    }
+    this.tokens.set(tokenConfig.uid, tokenConfig);
+    if (meta !== undefined) {
+      this.tokensMetadata.set(tokenConfig.uid, meta);
+    }
+  }
+
+  /**
+   * Iterate on registered tokens.
+   *
+   * @async
+   * @returns {AsyncGenerator<ITokenData & Partial<ITokenMetadata>>}
+   */
+  async *registeredTokenIter(): AsyncGenerator<ITokenData & Partial<ITokenMetadata>> {
+    for (const tokenConfig of this.registeredTokens.values()) {
+      const tokenMeta = this.tokensMetadata.get(tokenConfig.uid);
+      yield { ...tokenConfig, ...tokenMeta };
+    }
+  }
+
+  /**
+   * Register a token.
+   *
+   * Obs: we require the token data because the token being registered may not be on our storage yet.
+   *
+   * @param token Token config to register
+   * @async
+   * @returns {Promise<void>}
+   */
+  async registerToken(token: ITokenData): Promise<void> {
+    this.registeredTokens.set(token.uid, token);
+  }
+
+  /**
+   * Unregister a token.
+   *
+   * @param {string} tokenUid Token id
+   * @async
+   * @returns {Promise<void>}
+   */
+  async unregisterToken(tokenUid: string): Promise<void> {
+    this.registeredTokens.delete(tokenUid);
+  }
+
+  /**
+   * Edit token metadata on storage.
+   * @param {string} tokenUid Token id to edit
+   * @param {Partial<ITokenMetadata>} meta Metadata to save
+   * @returns {Promise<void>}
+   */
+  async editToken(tokenUid: string, meta: ITokenMetadata): Promise<void> {
```

Comment:
I think `editTokenMeta` would be better. What do you think?

Reply:
Same as `editAddressMeta`